test/src/com/xilinx/rapidwright/design/compare/TestDesignComparator.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2024, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Chris Lavin, AMD Research and Advanced Development.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.xilinx.rapidwright.design.compare;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.xilinx.rapidwright.design.Cell;
+import com.xilinx.rapidwright.design.Design;
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SiteInst;
+import com.xilinx.rapidwright.design.Unisim;
+import com.xilinx.rapidwright.device.Device;
+import com.xilinx.rapidwright.device.PIP;
+import com.xilinx.rapidwright.device.Site;
+import com.xilinx.rapidwright.device.SitePIP;
+import com.xilinx.rapidwright.device.SiteTypeEnum;
+import com.xilinx.rapidwright.support.RapidWrightDCP;
+import com.xilinx.rapidwright.util.Utils;
+
+/**
+ * Tests various design diffs using DesignComparator.
+ *
+ */
+public class TestDesignComparator {
+
+    private void compareDesign(int expectedTotalDiffs, int specificDiff, DesignDiffType type, 
+            DesignComparator dc, Design gold, Design test) {
+        int diffs = dc.compareDesigns(gold, test);
+        if (expectedTotalDiffs == 10)
+            dc.printDiffReport(System.out);
+        Assertions.assertEquals(expectedTotalDiffs, diffs);
+        Assertions.assertEquals(specificDiff, dc.getDiffList(type).size());
+    }
+
+
+    @Test
+    public void testDesignComparator() {
+        Design gold = RapidWrightDCP.loadDCP("picoblaze_2022.2.dcp");
+        Device device = gold.getDevice();
+
+        DesignComparator dc = new DesignComparator();
+        int diffs = dc.compareDesigns(gold, gold);
+        Assertions.assertEquals(0, diffs);
+
+        Design test = RapidWrightDCP.loadDCP("picoblaze_2022.2.dcp");
+        diffs = dc.compareDesigns(gold, test);
+        Assertions.assertEquals(0, diffs);
+
+        // Same device, different package
+        Design test2 = new Design(test.getName(), "xcvc1502-vsva2197-2MP-e-S");
+        test2.setNetlist(test.getNetlist());
+        SiteInst siteInst = null;
+        for (SiteInst si : test.getSiteInsts()) {
+            test2.addSiteInst(si);
+            if (Utils.isSLICE(si)) {
+                siteInst = si;
```

Comment:
Is this intended to find an arbitrary slice?

Reply:
I switched it to getting a specific instance since all that subsequent tests now depend on that specific site instance.